game/player.asm

```diff
@@ -0,0 +1,145 @@
+.section "Player" BANK 0 SLOT "ROM"
+
+nop
+
+.struct Player
+    oam_manager_ptr instanceof OAMManager
+    input instanceof Input
+    oam_obj_ptr dw ; Pointer to the requested OAM object
+    ; enabled db
+.endst
+
+.enum $0000
+    player instanceof Player
+.ende
+
+Player_Init:
+    pha
+    phy
+    phx
+
+    ; Init Input
+    call(Input_Init, player.input)
+    call(OAMManager_Init, player.oam_manager_ptr)
```

Comment:
The OAM Manager is a singleton and does not need to be re-initialized in your `Player_Init`. There is one OAM space physically and the manager is setup as part of the `Engine_Init` logic. You can delete this line.